src/main/java/org/websoso/WSSServer/repository/PopularFeedCustomRepositoryImpl.java

```diff
@@ -0,0 +1,44 @@
+package org.websoso.WSSServer.repository;
+
+import static org.websoso.WSSServer.domain.QBlock.block;
+import static org.websoso.WSSServer.domain.QPopularFeed.popularFeed;
+
+import com.querydsl.jpa.impl.JPAQueryFactory;
+import java.util.List;
+import java.util.stream.Stream;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Repository;
+import org.websoso.WSSServer.domain.PopularFeed;
+
+@Repository
+@RequiredArgsConstructor
+public class PopularFeedCustomRepositoryImpl implements PopularFeedCustomRepository {
+
+    private final JPAQueryFactory jpaQueryFactory;
+
+    @Override
+    public List<PopularFeed> findTodayPopularFeeds(Long userId) {
+        List<Long> blockingIds = jpaQueryFactory
+                .select(block.blockedId)
+                .from(block)
+                .where(block.blockingId.eq(userId))
+                .fetch();
+
+        List<Long> blockedIds = jpaQueryFactory
+                .select(block.blockingId)
+                .from(block)
+                .where(block.blockedId.eq(userId))
+                .fetch();
+
+        List<Long> blockIds = Stream.concat(blockingIds.stream(), blockedIds.stream())
+                .distinct()
+                .toList();
```

Comment:
q; 혹시! 이 세 친구들을 하나로 묶을 수는 없을까요? 요롷게는 안되려나요
```suggestion
private List<Long> findBlockedOrBlockingUserIds(Long userId) {
        return jpaQueryFactory
                .select(block.blockingId)
                .from(block)
                .where(block.blockingId.eq(userId)
                        .or(block.blockedId.eq(userId)))
                .fetch();
    }
```

Reply:
로직이 달라질 것 같은데요~!

Reply:
아니 그렇네요! 왜 저렇게 썼었지 제가 재시도를 해보겠습니다
```suggestion
List<Long> blockIds = jpaQueryFactory
        .select(block.blockingId)
        .from(block)
        .where(block.blockingId.eq(userId)
            .or(block.blockedId.eq(userId)))
        .union(
            jpaQueryFactory
            .select(block.blockedId)
            .from(block)
            .where(block.blockingId.eq(userId)
                .or(block.blockedId.eq(userId)))
        )
        .fetch();
```
이렇게는 어떠신가요? union으로 합치게 되면 distinct로 자동으로 적용되고 별개로 `Stream.concat()`같은 과정도 거칠 필요가 없어질 것 같아서요!

Reply:
`JPQL`이랑 `QueryDSL`에서는 `UNION`을 지원하지 않아서 공식문서에서 제시해준 방식으로 구현했는데,
혹시 `UNION`을 사용하는 방법이 있는건데 제가 못찾는건가요???....

아래는 References에 작성해둔 공식문서 링크입니다.
> https://github.com/querydsl/querydsl/blob/master/querydsl-sql/src/test/java/com/querydsl/sql/UnionBase.java

Reply:
`JPQL`이랑 `QueryDSL`에서는 UNION을 지원하지 않는 것으로 알고 있는데욥, JPAQueryFactory로 .union()을 사용하는 방법 레퍼런스 참고하신 거 있나욥!?

---

query-dsl 공식 github discussions에도 2021년 글이긴 한데 jpa에서는 지원하지 않는다고 나와있고, (https://github.com/querydsl/querydsl/discussions/2927)

query-dsl 공식 github에 jpa가 아닌 native sql로 사용하는 방식은 지원하고 있는 것 같습니다. -> com.querydsl.jpa가 아닌 com.querydsl.sql의 SQLQueryFactory를 사용(https://github.com/querydsl/querydsl/blob/8f96f416270d0353f90a6551547906f3c217833a/querydsl-sql/src/test/java/com/querydsl/sql/UnionBase.java#L73)

---

> JPQL이랑 QueryDSL 둘 다 UNION을 지원하지 않아서 native Query를 쓸까 하다가
native query가 dbms에 종속적이기도 하고(물론 dbms를 바꿀 것 같진 않긴 하지만,,),
타입 안정성을 챙길 수 없어서 QueryDSL을 선택했습니다.

SQLQueryFactory를 사용하고 SQLTemplates로 DBMS에 맞는 SQL 문법을 적용해서 사용할 수 있긴 할 것 같아요!

Reply:
아 그렇네요 그냥 union을 쓰면 편할 것 같다는 생각이었는데

> JPQL이랑 QueryDSL 둘 다 UNION을 지원하지 않아서 native Query를 쓸까 하다가

이걸 읽고 1초만에 까먹은 금붕어가 되었습니다.. 일단은 이대로 사용하는 것도 좋을 것 같네요!